docs/apache-airflow/howto/timetable.rst

```diff
@@ -0,0 +1,298 @@
+  example.
+* ``infer_data_interval``: When a DAG run is manually triggered (from the web
+  UI, for example), the scheduler uses this method to learn about how to
+  reverse-infer the out-of-schedule run's data interval.
+
+We'll start with ``infer_data_interval`` since it's the easier of the two:
+
+.. exampleinclude:: /../../airflow/example_dags/plugins/workday.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_timetable_infer_data_interval]
+    :end-before: [END howto_timetable_infer_data_interval]
+
+The method accepts one argument ``run_after``, a ``pendulum.DateTime`` object
+that indicates when the DAG is externally triggered. Since our timetable creates
+a data interval for each complete work day, the data interval inferred here
+should usually start at the midnight one day prior to ``run_after``, but if
+``run_after`` falls on a Sunday or Monday (i.e. the prior day is Saturday or
+Sunday), it should be pushed further back to the previous Friday. Once we know
+the start of the interval, the end is simply one full day after it. We then
+create a :class:`~airflow.timetables.base.DataInterval` object to describe this
+interval.
+
+Next is the implementation of ``next_dagrun_info``:
+
+.. exampleinclude:: /../../airflow/example_dags/plugins/workday.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_timetable_next_dagrun_info]
+    :end-before: [END howto_timetable_next_dagrun_info]
+
+This method accepts two arguments. ``last_automated_dagrun`` is a
+:class:`~airflow.timetables.base.DataInterval` instance indicating the data
+interval of this DAG's previous non-manually-triggered run, or ``None`` if this
+is the first time ever the DAG is being scheduled. ``restriction`` encapsulates
+how the DAG and its tasks specify the schedule, and contains three attributes:
+
+* ``earliest``: The earliest time the DAG may be scheduled. This is a
+  ``pendulum.DateTime`` calculated from all the ``start_date`` arguments from
+  the DAG and its tasks, or ``None`` if there are no ``start_date`` arguments
+  found at all.
+* ``latest``: Similar to ``earliest``, this is the latest time the DAG may be
+  scheduled, calculated from ``end_date`` arguments.
+* ``catchup``: A boolean reflecting the DAG's ``catchup`` argument.
+
+.. note::
+
+    Both ``earliest`` and ``latest`` apply to the DAG run's logical date
+    (the *start* of the data interval), not when the run will be scheduled
+    (usually after the end of the data interval).
+
+If there was a run scheduled previously, we should now schedule for the next
+weekday, i.e. plus one day if the previous run was on Monday through Thursday,
+or three days if it was on Friday. If there was not a previous scheduled run,
+however, we pick the next workday's midnight after ``restriction.earliest``
+(unless it *is* a workday's midnight; in which case it's used directly).
+``restriction.catchup`` also needs to be considered---if it's ``False``, we
```

Comment:
Sphinx renders triple-dash as an em-dash by default. This can be disabled, but from what I can tell we currently have it on.

https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-smartquotes

Reply:
emdash ftw. Nice typography makes me happy.

Reply:
Ah, cool, TIL. Do we want `<space>---<space>` then so we have spaces?

Reply:
I quickly searched and there doesn’t seem to be an established rule (both are generally accepted), so I’ll defer this to whatever the majority of reviewers think.

Reply:
The LaTeX author in me says no spaces, but I don't have a strong preference.